### PR TITLE
Adds the Nanotrasen Hospital Vessel Asclepius, updates admin shuttle

### PR DESCRIPTION
**Pull request by EvadableMoxie**
Adds the Nanotrasen Hospital Vessel Asclepius.  This ship can be spawned via the shuttle manipulator to replace the Admin shuttle.

The new 'admin' category on the shuttle manipulator:

shuttlemanip2

Often, the destruction of Medbay functionally ends a round, or worse, when the round is not ended dead players no longer have a way to reenter the round.  The Asclepius gives admins a tool to essentially restore the station's ability to tend to crew and get them back into the round.  This is useful for when they want to keep a round going, or for situations where evaccing isn't viable, such as terror spiders or blobs. As this is an administrative tool that can only be deployed by the admins, normal balance issues are less of a problem.  Even so, I have tried to make the ship fairly balanced by making it slightly inferior to the actual medbay in terms of resources.

The Asclepius can transit to the station and dock at the southern arrivals port.  Additionally, it is outfitted with a navigation computer which allows it to jump to custom locations on z1, z3, z4 and z5 (The Cyberiad, Telecomms,  Construction Site and Mining Asteroid).  ERT access is required to jump the ship.

The Asclepius also has a keycard authentication device and comms console, allowing it to function somewhat as a mobile command center, though lacking most of the terminals the bridge does.  It lacks weaponry, though it does have two wall chargers in the command center for recharging the weapons of the ERT medics that crew it.

The ship itself is a functional medbay, featuring a cloning room with a cyrotube, two operating rooms, a chemistry room with 2 chem setups and a general area with 2 cryotubes and 2 sleepers.  There is a Pan.D.E.M.I.C 2200 in the command center for producing vaccines from anti-bodies, but it has none of the equipment for developing viruses that the Cyberiad does.  It also completely lacks a genetics department. There is no wine for producing SR (ERT medics have omni), so the crew will need to scavenge for it. All of the equipment uses stock parts.  If the crew want upgrades they'll need to get them from science.

In terms of equipment, the ship has 2 Nanomed vendors, IVs, Blood, Medkits, Radiation suits, bio suits, two medical hardsuit storage units (CMO access required), and some general equipment such as medical belts and basic health huds.  There is one wall mounted defib unit in the main area, and 1 normal defib in each of the ORs.  The main area has a cell charger for recharging the defib units. The external windows have shutters, and the airlocks have blast doors. These can be toggled in the command center using ERT access.  Overall, this is a bit less than what the actual medbay gets.

The chemistry room and operating rooms require Chemistry and Medical Doctor access respectively. The command center requires Bridge access. The ship itself, cloning, and the main treatment area do not require access. It's expected that the Cyberiad crew will help man the ship, with a few ERT medics to assist, hence the access requirements.

The southern airlock at arrivals has been extended 1 tile west so the ship can dock without blocking disposals.  The dock at CentCom has been extended 1 tile east and 2 tiles south.  This does not interfere with the emergency shuttle docking.

The shuttle has cameras installed on a separate network. The AI can access this network at all times, but as the shuttle is surrounded by walls while at CC it cannot see into any other area of CC while the shuttle is docked in the default location.

**Images of sprite/map changes (IF APPLICABLE):**

The NHV Asclepius:

NHVAscl

Arrivals southern docking port, after map changes:

AsclDockingAfter

Centcom docking port, after map chances:

Asclcentcomdockingafter

The admin shuttle has also gotten a bit of love.  It now has a name in game, uses a navigation computer to jump to specific locations, and access requirements have been changed from admin to ert.  Tiny fans have been added to the airlock to support the custom location functionality. 

The updated admin shuttle:

argos

Note: This is my first PR so please let me know what I am doing wrong.

**Changelog:**
:cl:
add: Added the Nanotrasen Hospital Vessel Asclepius, able to be deployed by admins.
tweak: Admin shuttle now requires ERT clearance instead of admin clearance.
tweak: Admin shuttle now has a navigation computer and tiny fans at the airlock. 
/:cl:

### CODE REVIEW

**Comment by Couls** [comment]
One word: Yes

**Comment by Kyep** [comment]
I like this.

Maybe store the old admin shuttle template in the templates system. You can access it using the holographic shuttle template loader in the middle of the tables in the ERT spawn area. That way if admins need the old template for some reason, they can still load it - and potentially other admin-only shuttles could be added in future using that large shuttle port.

**Comment by EvadableMoxie** [comment]
I can look into the shuttle template system, although my knowledge of the code is limited, so no promises!

**Comment by Ty-Omaha** [comment]
Good idea but it should have a template manager so we're not locked to a hospital ship.

**Comment by datlo** [comment]
I cant see it if done on the screenshot but the ship needs either tiny fans under the outside airlocks or a cycling airlock like the nuke ops ship (the latter being preferable since its a large shuttle)

Otherwise this will cause atmos issues out the wazoo whenever you set a custom destination since those arent at a station dock

**Comment by Spacemanspark** [comment]
Echoing others, this should be a template.

Other than that, quite impressive.
:+1:

**Comment by EvadableMoxie** [comment]
> I cant see it if done on the screenshot but the ship needs either tiny fans under the outside airlocks or a cycling airlock like the nuke ops ship (the latter being preferable since its a large shuttle)

It has tiny fans, airlocks are just really awkward.

As for templates... I guess I'll work on it.  I have no idea what I'm doing so this'll be interesting.

**Comment by Kyep** [comment]
When I added a new escape shuttle in https://github.com/ParadiseSS13/Paradise/pull/7504/files it only required adding a small additional block to code/datums/shuttles.dm. While this might require slightly more (unsure if admin shuttle is in the system) that might be a good place to start.

**Comment by EvadableMoxie** [comment]
From what I can tell, adding new shuttle templates for the emergency shuttle is pretty easy because you're just adding a new entry to a list.  But this is something different, I'm not adding a new template to the emergency shuttle, I'm trying to create an entirely new category of ship for a new docking port.  The fact that you can edit both the ferry and emergency shuttle gives me hope that the system is robust enough to add a third type of ship.  

I just have no idea how to do that.

**Comment by Kyep** [comment]
Well,  have you tried defining a new type, such as /datum/map_template/shuttle/admin then clicking the map loader hologram in the ERT spawn and seeing what happens? 
https://github.com/ParadiseSS13/Paradise/blob/f8d88f25b8c4574990a6f6170b85070cf990d8b6/code/modules/shuttle/shuttle_manipulator.dm seems to define how the shuttle_manipulator works... you could probably get some clues about what it expects to find by looking at that.

**Comment by EvadableMoxie** [comment]
I've made some progress on it but I'm at a roadblock. 

shuttlemanip

The 'ship' button is new, it's the category for the admin shuttle and hospital ship. The problem is when I click it, it opens the emergency list, not the ships list.  I'm not sure what it's doing that. I'm not sure if it's a UI issue or I haven't formatted something correctly.

**Comment by Fox-McCloud** [comment]
I see a hugbox finally managed to acquire engines, a cockpit, and wings.

I admit, I'm not a fan of features that cling on to the notion that the round absolutely has to go on for the longest time possible/as close to 2 hours as possible.

**Comment by Spacemanspark** [comment]
Is the first bit of that comment really necessary @Fox-Mccloud

That said, I can see the concern... but it's also an admin only feature. This isn't likely to be something that will be used very often; merely if we feel like the round _should_ continue.

**Comment by Kyep** [comment]
I got your stuff largely working with the shuttle system.
Here's the version I got working: https://github.com/ParadiseSS13/Paradise/compare/master...Kyep:admin_shuttle_test

Here's what I did to make this work:
1) I avoided changing any paths unless I absolutely had to.  You changed /area/shuttle/administration to /area/shuttle/asclepius, /obj/machinery/computer/shuttle/admin to /obj/machinery/computer/shuttle/asclepius, etc. These sorts of path changes make things harder than they need to be. It is the admin shuttle, not the asclepius. The admin shuttle can be one of several ships, depending on what is loaded.
2) You have multiple pipes or pipe-connecting things in the same tile, such as a pipe and a freezer. This causes runtimes and map load failures. My version of your template has all the pipes deleted, which resolves this. If you use my versions, you'll need to re-add the pipes.
3) I dropped support for the advanced shuttle docking computer that lets you dock anywhere. Not because it was necessarily broken, but just in the interests of quickly seeing if I could get the basics working.
4) I made sure that my admin_base.dmm and admin_frigate.dmm files did not contain a stationary docking port, and their mobile docking port had 'timid' var set to 1. It fails with an error about not being timid if this is not done.
5) I changed the height/width/dheight/dwidth of all the docking ports involved, to ensure that it would account for both sizes of shuttle. In retrospect my values probably aren't perfect - they're arguably too big for the small standard red shuttle. They do, however, cover both and fit into the new alcove on z2.

If you want to continue from my version, you'll need to re-add the advanced shuttle docking computer, pipes, and possibly other things.

**Comment by EvadableMoxie** [comment]
> I admit, I'm not a fan of features that cling on to the notion that the round absolutely has to go on for the longest time possible/as close to 2 hours as possible.

That's pretty hyperbolic. An admin tool that can be used in certain situations to extend certain rounds is not the same as advocating for extended every round. This is essentially a different from of ERT or Gamma, resources the admins can choose to provide to the station when appropriate.  It isn't meant to happen every round.

**Comment by EvadableMoxie** [comment]
By popular demand: The admin shuttle and hospital ship are now handled by the shuttle manipulator. The admin ship is the default ship, and the hospital ship can be spawned in it's place.  You can swap back between the two freely, however once you put one in play you should generally not swap out to a different one.  Swapping template while the ship is in use will delete everyone on the ship, as the system wasn't really designed for hotswapping.  Additionally, if you use a custom docking location and then switch ships you'll be unable to use custom docking with any other ship for the rest of the round.

The take away here is this: Decide which ship you want to use for this round, spawn that one, and then don't touch it.

Additionally, the admin shuttle has gotten some love.  It now requires ERT access instead of admin access, has a custom navigation computer, and has tiny fans at the airlocks. This should allow admins to give it to players for use for spec ops.   The airlocks at CC are still admin only.

The new 'admin' category on the shuttle manipulator:

shuttlemanip2

The updated Argos

argos

**Comment by Fox-McCloud** [comment]
This wasn't mapped properly.

you need to actually mapmerge your things.

**Comment by EvadableMoxie** [comment]
> From what I see some of the map changes are in the old format instead of the new one. I need to confirm with others if this is OK.

Thanks for taking the time to reply.  I'm not sure what the new format is versus the old.  If it does need to be updated is there any documentation anywhere on what the proper way to do it is?

**Comment by variableundefined** [comment]
Unfortunately I don't really have any knowledge of how the two map formats really work. I think Fox or Arc might know something about it.

TGM's format have each object on separate line while the old format has every object on a tile on the same line.

**Comment by variableundefined** [comment]
This would be good to go once it is in correct format.

**Comment by EvadableMoxie** [comment]
Is the issue with the cyberiad and z2 changes, or with the new map files admin_admin and admin_hospital? I didn't map merge them as they are new map files so I was under the impression I didn't have to.

**Comment by EvadableMoxie** [comment]
Okay, the templates should be formatted properly, if that was the issue.  Let me know if there are any other issues.

**Comment by variableundefined** [comment]
Looks good, but it is conflicted at the moment.

File: code/modules/events/event_manager.dm [hunk 1]
```diff
@@ -328,6 +328,7 @@
 	/area/shuttle/specops/station,
 	/area/shuttle/prison/station,
 <<<<<<< HEAD
+<<<<<<< HEAD
 <<<<<<< HEAD
```
Comment (taukausanake) [inline code/modules/events/event_manager.dm]: _Internal screaming._ How many changes did you make at once?

**Comment by EvadableMoxie** [comment]
At this point it's just way too conflicted and has just way too many commits.  I'm going to start over from scratch which should be much, much easier. Someone who knows what they're doing could probably just fix it, but it's beyond me.  I still have the map files and the important code changes so it won't take long for me to redo this.

**Comment by taukausanake** [comment]
Wish you the best for this. I really like the idea of a medical shuttle

**Pull request closed without merging**